shotover/src/unix_socket_server.rs

```diff
@@ -0,0 +1,128 @@
+use crate::hot_reload::{Request, Response};
+use serde_json;
+use std::collections::HashMap;
+use std::path::Path;
+use tokio::io::{AsyncBufReadExt, AsyncWriteExt, BufReader};
+use tokio::net::{UnixListener, UnixStream};
+
+pub struct UnixSocketServer {
+    socket_path: String,
+    listener: Option<UnixListener>,
+}
+
+impl UnixSocketServer {
+    pub fn new(socket_path: String) -> Self {
+        Self {
+            socket_path,
+            listener: None,
+        }
+    }
+
+    pub async fn start(&mut self) {
+        if Path::new(&self.socket_path).exists() {
+            std::fs::remove_file(&self.socket_path).unwrap();
+        }
+        let listener = UnixListener::bind(&self.socket_path).unwrap();
+        self.listener = Some(listener);
+    }
+
+    pub async fn run(&mut self) {
+        let listener = self.listener.as_ref().unwrap();
+        loop {
+            let (stream, _) = listener.accept().await.unwrap();
+            self.handle_connection(stream).await;
```

Comment:
Your current design seems to use a unique connection per request/response.
This is definitely unusual, but could actually be a neat idea!

Usually a client will open a single connection and then the client and server will exchange many request/response pairs over the lifetime of the connection.
This is efficient since we reuse the connection which has a cost to setup.

However, for our use case, efficiency is less important than correctness.
And using one connection per request/response is simpler and therefore easier to get right.

So I'm happy to proceed with your current approach.